kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java

```diff
@@ -251,8 +265,13 @@ public AssetConfiguration getAssetConfiguration() {
         return this.assetConfiguration;
     }
 
-    public void setDriver(Driver driver) {
+    public synchronized void setDriver(Driver driver) {
         this.driver = driver;
+        if (driver != null) {
+            List<DriverRecord> readRecords = getAllReadRecords();
+            hasReadChannels = !readRecords.isEmpty();
+            tryPrepareRead(readRecords);
```

Comment:
Is there any specific reason why the driver's prepared read is getting triggered while injecting the Driver instance?